context/TodoContext.tsx

```diff
@@ -0,0 +1,37 @@
+import { TodoType } from '@/helpers/common/types/todoType'
+import {
+  ReactNode,
+  SetStateAction,
+  createContext,
+  useState,
+  Dispatch,
+  useMemo,
+} from 'react'
+
+type TodoContextState = {
+  todos: TodoType[]
+  setTodos: Dispatch<SetStateAction<TodoType[]>>
+}
+
+const initialState: TodoContextState = {
+  todos: [],
+  setTodos: () => {},
+}
+
+export const TodoContext = createContext<TodoContextState>(initialState)
+
+type TodoContextProviderPropsType = {
+  children: ReactNode
+}
+
```

Comment:
This is cool that you are familiar with context.   I think once you became more familiar with the capabilities of react query, you might not think this is necessary, but lets talk about it.  overall the code looks nice so I think it's just a matter of knowing more about react query's capabilities.